src/core/shader.js

```diff
@@ -45,6 +45,8 @@ Shader.prototype = {
       'gl_FragColor = vec4(1.0, 0.0, 1.0, 1.0);' +
     '}',
 
+  glslVersion: null,
```

Comment:
Depends on if we want to support WebGL1 or not. (Though I'm not sure if A-Frame works with a WebGL1 context at the moment as I've never tried it)

Reply:
This should not be part of the public API. Should check if the [WebGLRenderer is webgl1 or not](https://github.com/mrdoob/three.js/blob/95fbc39b07f677a3f550f185dd2d3d18d4a52c93/src/renderers/WebGL1Renderer.js#L5). Still I think is moot because as you said A-Frame doesn't offer a WebGL1 option

Reply:
This definitely depends of decision to support WebGL1. If we drop WebGL1 support than sdf/msdf code should also be stripped/cleaned to remove GL1 shaders (which are differs from GL2 ones) and some startup checks for WebGL version has to be added to warn user of unsupported WebGL version.

Reply:
This THREE functionality looks a bit not clean as it is THREE who injects extra code in to shader so it is THREE job to inject code right way instead of forcing users to fill in the additional version parameter

Reply:
> This should not be part of the public API. Should check if the [WebGLRenderer is webgl1 or not](https://github.com/mrdoob/three.js/blob/95fbc39b07f677a3f550f185dd2d3d18d4a52c93/src/renderers/WebGL1Renderer.js#L5).

It's a bit more subtle than that. The `WebGLRenderer` supports both WebGL 1 and WebGL 2 and will automatically pick one. The `WebGL1Renderer` class only exists to _force_ WebGL 1. In other words, just checking `.isWebGL1Renderer` isn't sufficient, as you can have a normal `WebGLRenderer` with a WebGL 1 context (in case your browser/device doesn't support WebGL 2).

What we're discussing however is the **glslVersion**. This is tied to the WebGL version in the sense that you can only use ES 3.0 shaders with WebGL 2, but you can use ES 1.0 shaders on both WebGL 1 and 2. There are differences in syntax so you can't simply tag on a `#version 300 es` on an ES 1.0 shader.

So if we hide this and default to `THREE.GLSL3` we're basically deciding two things at the same time:

- A-Frame will only supports WebGL2 (which it might already depend in other areas of the code?)
- A-Frame will only supports ES 3.0 shaders

Which I don't view as a problem, but is worth having clear in case someone wants to incorporate a (raw) ES 1.0 shader. And as @nightgryphon points out, if we do opt for this approach we should cleanup the sdf/msdf code and add an explicit WebGL2 check.

Reply:
> This THREE functionality looks a bit not clean as it is THREE who injects extra code in to shader so it is THREE job to inject code right way instead of forcing users to fill in the additional version parameter

Totally agree with you on this. I don't mind THREE.js injecting things into shaders, but it's surprising that they now also do this for `RawShaderMaterial`...